ctest/CMakeLists.txt

```diff
@@ -50,9 +50,29 @@ configure_file(${CMAKE_CURRENT_LIST_DIR}/CTestCustom.cmake ${CMAKE_BINARY_DIR} C
 # Set environement variables so that executables and libraries can be find by tests.
 # (avoids a make install before make test)
 function(set_test_env TestName)
+    # A cmake PROPERTY value is just a string.
+    # The environment property is implemented as a semicolon separated
+    # list of value pair. Each value pair being separated by `=`
+    # Eg: the python env equivalent of `{"A":"1", "B":"/;/bin"}` is `A=1;B=/\;/bin`
+    #
+    # As a consequences cmake can not differentiate between a `;` that separates
+    # two environment variable and a `;` contained in a variable.
+    #
+    # Thus all `;` in the environment variable must be escaped.
```

Comment:
Thus all `;` in environment variables must be escaped before inserting them in the ENVIRONMENT PROPERTY.